framework/src/play/cache/CacheImpl.java

```diff
@@ -1,14 +1,21 @@
 package play.cache;
 
+import org.apache.commons.lang3.NotImplementedException;
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.Properties;
 
 /**
  * A cache implementation. expiration is specified in seconds
  *
  * @see play.cache.Cache
  */
 public interface CacheImpl {
+  static CacheImpl instance(@Nonnull Properties playProperties) throws IOException {
+    throw new NotImplementedException();
+  }
+
```

Comment:
@cies I don't think this method is needed. In Java, static methods are not inherited by subclasses.

Reply:
@asolntsev  The `instance(Props)` method part of the singleton pattern that I want to have enforced on all implementations. I've put this method here to show to implementing devs that they need to implement this method.

Also I call this method on through the interface (and reflection) here in `Cache.java`:

```java
        // Since it implements the `CacheImpl` interface, it should have a static `instance` method.
        Method method = cacheImplClass.getDeclaredMethod("instance", Properties.class);
        cacheImpl = (CacheImpl) method.invoke(null, Play.configuration);
```

I'm not sure if it is actually needed, but it does point implementing devs in the right direction and and allow me to call it through reflection with some certainty.

Based on your comment I prefer to keep it. Maybe I do not understand it well enough though.

Reply:
> allow me to call it through reflection with some certainty

@cies For sure, this method is not needed. It doesn't add any certainty. :)

You can give a hint to developers just in javadoc.

Reply:
Now explained with javadoc as per you request.